File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;
import entity.Result;

import java.util.List;
import java.util.Map;

/**
 * 品牌的接口
 */
public interface BrandService {
    /**
     * 返回所有的品牌
     * @return
     */
    List<TbBrand> findAll();

    /**
     * 返回分页的数据
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(int pageNum,int pageSize);

    /**
     * 增加品牌
     * @param tbBrand
     */
    void add(TbBrand tbBrand);

    /**
     * 根据id获取品牌
     * @param id
     * @return
     */
    TbBrand findOne(Long id);

    /**
     * 修改品牌
     * @param tbBrand
     */
    void update(TbBrand tbBrand);

    /**
     * 根据id
     * @param ids 删除数据
     */
    Result delete(Long[] ids);

    /**
     * 模糊查询并且分页
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult findPage(TbBrand tbBrand,int pageNum,int pageSize);

    /**
     * 下拉列表展示数据
     */
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
    //定义方法绑定数据 查询所有
    this.findAll=function(){
        //调用后台的数据
       return  $http.get('../brand/findAll.do');
    }
    //抽取分页
    this.findPage=function(pageNum,pageSize){
        return $http.get('../brand/findPage.do?pageNum='+pageNum+'&pageSize='+pageSize);
    }
    //根据id查询
    this.findOne=function(id){
        return $http.get('../brand/findOne.do?id='+id);
    }
    //抽取add
    this.add=function(entity){
        return $http.post('../brand/add.do',entity);
    }
    //抽取更新的update
    this.update=function(entity){
        return $http.post('../brand/update.do',entity);
    }
    //抽取删除的方法
    this.dele=function(ids){
        return $http.get('../brand/delete.do?ids='+ids);
    }
    //抽取模糊查询并且分页
    this.search=function(pageNum,pageSize,entity){
        return $http.post('../brand/search.do?pageNum=' + pageNum
            + '&pageSize=' + pageSize, entity)
    }
    //展示下拉列表
    this.selectOptionList=function(){
        //向后端获取数据
        return $http.get('../brand/selectOptionList.do');
    }
});